### ISSUE TITLE
Data classes with private properties not handled by encoding

### ISSUE THREAD

**Issue by jlkeesey** [valderman/4koma#7]
If you have a data class with a private field and use it in encoding (like in decodeWithDefaults()) it will fail with an IllegalAccessException. This is true whether the property has a backing field or not. The below code reproduces this problem. If the commented out line is uncommented encoding of the default value will fail. I don't know if this never came up or something changed in the way that Kotlin handles properties but it fails under the more recent Kotlins. I tested with under Kotlin v1.6.21. 

In checking the code I see that the encode() method tries to encode all properties of the data class. However, the decode() process only ever uses the primary constructor so encoding any value not used in the constructor is unnecessary as the value will not be used. 

I have a change which fixes this by only encoding the primary constructor properties. I will submit the PR as well.

```
import cc.ekblad.toml.model.TomlValue
import cc.ekblad.toml.serialization.from
import cc.ekblad.toml.tomlMapper

data class Config(
    val flag: Boolean = true,
    val text: String = "A Test",
) {
    // If the next line is uncommented, decode fails
    // private val combined: String = "$text $flag"
}

fun main() {
    val mapper = tomlMapper {}
    mapper.decodeWithDefaults(Config(), TomlValue.from(""))
}
```

**Comment by valderman** [comment]
Yep, this one slipped through the cracks. Kotlin didn't change anything, I messed up a bit there. Good catch!

**Comment by jlkeesey** [comment]
I don't have permission to connect this issue with the PR #8
